src/services/layers/layers.service.ts

```diff
@@ -18,54 +18,66 @@ export class LayersServices {
     }
   }
 
-  toggleLayer = (id: number, show = true) => {
-    const layer = themesService.findById(id)
+  initLayer(layer: Layer) {
+    layer.opacity = layer.previousOpacity = layer.metadata?.start_opacity ?? 1
+    return layer
+  }
+
+  handleExclusionLayers(layer: Layer) {
```

Comment:
I did not dive into the logic here, but if I understand right the changes are meant to move the methods from the `state` to the `service`?

Reply:
Indeed, layer must not be initialize in a state manager, this should be handled by the layer service.

Reply:
state is just about reducers, no logic in there.